HyperVExtension/src/DevSetupEngine/ConfigurationFileHelper.cs

```diff
@@ -1,198 +1,201 @@
-                Logging.Logger()?.ReportCritical(sourceComponent, diagnosticInformation.Message);
-                return;
-            case WinGet.DiagnosticLevel.Verbose:
-            case WinGet.DiagnosticLevel.Informational:
-            default:
-                Logging.Logger()?.ReportInfo(sourceComponent, diagnosticInformation.Message);
-                return;
-        }
-    }
-
-    /// <summary>
-    /// Convert a string to an input stream
-    /// </summary>
-    /// <param name="str">Target string</param>
-    /// <returns>Input stream</returns>
-#pragma warning disable CA1859 // Use concrete types when possible for improved performance
-    private IInputStream StringToStream(string str)
-#pragma warning restore CA1859 // Use concrete types when possible for improved performance
-    {
-        InMemoryRandomAccessStream result = new();
-        using (DataWriter writer = new(result))
-        {
-            writer.UnicodeEncoding = UnicodeEncoding.Utf8;
-            writer.WriteString(str);
-            writer.StoreAsync().AsTask().Wait();
-            writer.DetachStream();
-        }
-
-        result.Seek(0);
-        return result;
-    }
-}
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Buffers;
+using Microsoft.Management.Configuration;
+using Microsoft.Management.Configuration.Processor;
+using Microsoft.Management.Configuration.SetProcessorFactory;
+using Windows.Storage.Streams;
+using WinRT;
+using DevSetupEngineTypes = Microsoft.Windows.DevHome.DevSetupEngine;
+using WinGet = Microsoft.Management.Configuration;
+
+namespace HyperVExtension.DevSetupEngine;
+
+/// <summary>
+/// Helper for applying a configuration file. This exists so that we can
+/// use it in an elevated or non-elevated context.
+/// </summary>
+public class ConfigurationFileHelper
+{
+    public class ApplicationResult
+    {
+        public WinGet.ApplyConfigurationSetResult Result
+        {
+            get;
+        }
+
+        public bool Succeeded => Result.ResultCode == null;
+
+        public bool RequiresReboot => Result.UnitResults.Any(result => result.RebootRequired);
+
+        public Exception ResultException => Result.ResultCode;
+
+        public ApplicationResult(WinGet.ApplyConfigurationSetResult result)
+        {
+            Result = result;
+        }
+    }
+
+    private WinGet.ConfigurationProcessor? _processor;
+    private WinGet.ConfigurationSet? _configSet;
+
+    public ConfigurationFileHelper()
+    {
+    }
+
+    /// <summary>
+    /// Open configuration set from the provided <paramref name="content"/>.
+    /// </summary>
+    /// <param name="content">DSC configuration file content</param>
+    private ConfigurationResultTypes.OpenConfigurationSetResult OpenConfigurationSet(string content)
+    {
+        try
+        {
+            var modulesPath = Path.Combine(AppContext.BaseDirectory, @"runtimes\win\lib\net8.0\Modules");

```

Comment:
This is where the functional change is.
(Almost) everything else is due to EOL fixup.